### ISSUE TITLE
Feature Request: 'Codequery-Build' cmd triggers (user-modifiable) script to build cq.db...

### ISSUE THREAD

**Issue by gwiede** [ruben2020/codequery4vscode#7]
Instead of requiring the user to manually launch all the cmds that aggregate and assemble the cq.db database for CodeQuery to use, I propose the plugin install the following user-modifiable scripts:

- cscope-build.sh
- ctags-build.sh
- cq-build.sh

cq-build.sh:
```
#!/bin/sh
cd ${CQ_WS_DIR}   # = <workspace>/.vscode/codequery
exec ${CQ_PLUGIN_BIN}/cscope-build.sh
exec ${CQ_PLUGIN_BIN}/ctags-build.sh
${CQ_PLUGIN_BIN}/cqmakedb -s ${CQ_WS_DIR}/cq.db -c ${CQ_WS_DIR}/cscope.out -t ${CQ_WS_DIR}/tags -p
```
cscope-build.sh:
```
#!/bin/sh
find ${WS} -iname "*.c"    > ${CQ_WS_DIR}/cscope.files
find ${WS} -iname "*.cpp" >> ${CQ_WS_DIR}/cscope.files
find ${WS} -iname "*.cxx" >> ${CQ_WS_DIR}/cscope.files
find ${WS} -iname "*.cc " >> ${CQ_WS_DIR}/cscope.files
find ${WS} -iname "*.h"   >> ${CQ_WS_DIR}/cscope.files
find ${WS} -iname "*.hpp" >> ${CQ_WS_DIR}/cscope.files
find ${WS} -iname "*.hxx" >> ${CQ_WS_DIR}/cscope.files
find ${WS} -iname "*.hh " >> ${CQ_WS_DIR}/cscope.files
cscope -cbf ${CQ_WS_DIR}/cscope_c.out

# Uncomment below if using pycscope
# find ${WS} -iname "*.py"    > ${CQ_WS_DIR}/pycscope.files
# pycscope -i ${CQ_WS_DIR}/pycscope.files -f ${CQ_WS_DIR}/cscope_py.out

# Uncomment below if using starscope
# find ${WS} -iname "*.rb"    > ${CQ_WS_DIR}/cscope.files
# find ${WS} -iname "*.go"    >> ${CQ_WS_DIR}/cscope.files
# find ${WS} -iname "*.js"    >> ${CQ_WS_DIR}/cscope.files
# starscope -e cscope,${CQ_WS_DIR}
```

ctags-build.sh:
```
#!/bin/sh
ctags --fields=+i -n -L ${CQ_WS_DIR}/cscope.files -f ${CQ_WS_DIR}/tags
```

Hopefully this is enough to give you a rough idea of what I mean.  The key is that the user can adjust locations, etc. as needed to integrate with other tools and systems they might be using, etc.

**Comment by ruben2020** [comment]
I need to think about this more carefully, because this extension supports multiple programming languages  not just C/C++, and also multiple OS platforms including Windows. So, we need many many scripts.

**Comment by ruben2020** [comment]
@gwiede Feature implemented in codequery4vscode version 0.8.1. You've got to write your own rebuild script and put it in a standard place, and it can be called by the extension through the command palette.
